### PR TITLE
API refactoring

### LINKED ISSUE #32: Optimize the code for library usage
### Please describe your feature request:
The logic should be moved within an openrisk package and exposed via struct instance methods (as generally, it allows easy reuse within other projects). Here is an example skeleton:

```
package openrisk

type Options struct {
   ApiKey string
}

Type OpenRisk struct {}

func New(options *Options) (*Openrisk,error) { ... }

func (o *Openrisk) func GetCompletion(..) .. {...}
func (o *Openrisk) func MakeRequest(..) .. {...}
...
```

Usually, the main entry point is located in `cmd/openrisk/main.go`

### PR DESCRIPTION
**Pull request by dogancanbakir**
- [X] Expose an API for parsing scan result files (MD, JSONL, RAW)
- [X] Expose an API for generating a score for a single host/scan
- [x] [Expose an API for generating a score per multiple host/scan](https://github.com/projectdiscovery/openrisk/issues/7)
- [X] Move main entry point to cmd/openrisk
- [x] Add use as a library example

Closes #32, #7 and #30

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -21,10 +21,14 @@ jobs:
...
+        working-directory: cmd/openrisk
 
       - name: Test
         run: go test ./...
-        working-directory: .
+        working-directory: cmd/openrisk
+
+      - name: Running example
+        run: go run .
+        working-directory: examples/
```
Comment (ehsandeep) [inline .github/workflows/build-test.yml:32]:
```yaml
        env: 
          OPENAI_API_KEY: "${{ secrets.OPENAI_API_KEY }}"
```

This line can be added in the workflow to read the test token.

**Review by Mzack9999 — APPROVED** [review]
lgtm!

**Review by ehsandeep — COMMENTED** [review]
@dogancanbakir is https://github.com/projectdiscovery/openrisk/issues/7 limited to use as API and not possible with CLI? if that's the case, we can create follow up ticket to support the same via CLI as well.

**Review by ehsandeep — COMMENTED** [review]
@dogancanbakir is https://github.com/projectdiscovery/openrisk/issues/7 limited to use as API and not possible with CLI? if that's the case, we can create follow up ticket to support the same via CLI as well.

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**